### PR TITLE
Get distance from normalize vector

### PR DESCRIPTION
**Pull request by GuyAv46**
Get distance from normalize vector without normalizing it again.

### CODE REVIEW

File: src/VecSim/algorithms/brute_force/brute_force.cpp [hunk 1]
```diff
@@ -150,14 +150,17 @@ int BruteForceIndex::deleteVector(size_t label) {
 }
 
 double BruteForceIndex::getDistanceFrom(size_t label, const void *vector_data) {
+    return BruteForceIndex::getDistanceFrom(label, vector_data, true);
```
Comment (alonre24) [inline src/VecSim/algorithms/brute_force/brute_force.cpp:153]: Like on the syntactic sugar ;)

File: src/VecSim/vec_sim.cpp [hunk 2]
```diff
@@ -45,6 +45,13 @@ extern "C" int VecSimIndex_DeleteVector(VecSimIndex *index, size_t id) {
 extern "C" double VecSimIndex_GetDistanceFrom(VecSimIndex *index, size_t id, const void *blob) {
     return index->getDistanceFrom(id, blob);
 }
+extern "C" double VecSimIndex_GetDistanceFrom_withFlag(VecSimIndex *index, size_t id,
+                                                       const void *blob, bool normalize) {
+    return index->getDistanceFrom(id, blob, normalize);
+}
+extern "C" void VecSimIndex_PrepareVector(VecSimIndex *index, const void *source, void *dest) {
+    index->prepareVector(source, dest);
+}
```
Comment (alonre24) [inline src/VecSim/vec_sim.cpp:54]: I don't like the name `PrepareVector`... I think that it is better to just call the child at it's name - `NormalizeVector`.
Also, I don't think that this function should be a member of `VecSimIndex`, as it is more of a "static" function. Then the signature may become something like:
 `VecSim_NormalizeVector(const void *source, void *dest, size_t dim, VecSimType type)`
@DvirDukhan WDYT?

File: src/VecSim/vec_sim.h [hunk 3]
```diff
@@ -55,6 +55,30 @@ int VecSimIndex_DeleteVector(VecSimIndex *index, size_t id);
...
+ * @param id the id of the vector in the index.
+ * @param blob binary representation of the second vector. Blob size should match the index data
+ * type and dimension.
+ * @param normalize flag for skipping normalization. useful when you intend to use this function
+ * many times, and your blob is already normalized.
+ * @return The distance (according to the index's distance metric) between `blob` and the vector
+ * with id `id`.
+ */
+double VecSimIndex_GetDistanceFrom_withFlag(VecSimIndex *index, size_t id, const void *blob,
+                                            bool normalize);
```
Comment (alonre24) [inline src/VecSim/vec_sim.h:71]: When `normalize=True`, this API is equivalent to `VecSimIndex_GetDistanceFrom`, right? 
So why not calling it `VecSimIndex_GetDistanceFromNormalizedVector` (which means that we assume that `blob` is normalised) and avoid the `normalize` flag?

**Review by alonre24 — COMMENTED** [review]
It looks OK, go over my comments and suggestions and let's discuss it afterwards

File: src/VecSim/vec_sim.cpp (hunk 2 above)
Comment (GuyAv46) [reply src/VecSim/vec_sim.cpp:54]: the idea was to use this function before using `getDistanceFrom_withFlag` without a need to check what type of space we have.
sometimes this function will not normalize so calling it like that will be misleading.

File: src/VecSim/vec_sim.h (hunk 3 above)
Comment (GuyAv46) [reply src/VecSim/vec_sim.h:71]: this way Ill need to write the function twice or use helper func. not a big deal just another approach :). 
@DvirDukhan ?

**Pull request closed without merging**